src/app/_api/functions/recalculate-builder-score.ts

```diff
@@ -0,0 +1,40 @@
+"use server";
+
+import { revalidatePath, revalidateTag } from "next/cache";
+import { supabase } from "@/db";
+import { rollbarError } from "@/services/rollbar";
+import { getCurrentUser } from "../data/users";
+import { resyncPassportForUser } from "../external/talent-protocol";
+import { CacheKey } from "../helpers/cache-keys";
+
+export const recalculateBuilderScore = async (): Promise<number> => {
+  const user = await getCurrentUser();
+  if (!user) return 0;
+
+  const highestPassport = await resyncPassportForUser(user);
+  await supabase
+    .from("users")
+    .update({
+      passport_id: highestPassport.passport_id,
+      passport_builder_score: highestPassport.score,
+    })
+    .eq("id", user.id);
```

Comment:
It's probably best to add a `throwOnError` here. We are otherwise allowing this function to continue even if the update fails.